lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -955,6 +1063,26 @@ def get_vm_ide_device(self, vm=None):
 
         return None
 
+    def get_vm_sio_device(self, vm=None):
```

Comment:
Can you make this as a generic function which can be used by other functions like `get_vm_cdrom_device` and `get_vm_ide_device` and others ?

```python
def get_vm_device(self, vm=None, device_type=None):
    if not all([vm, device_type)]:
        return None
    for device in vm.config.hardware.device:
        if isinstance(device, device_type):
            return device
    return None
```

Reply:
Is this something we want in a "adding floppy support" PR vs a generic code cleanup PR afterwards?

Reply:
for the record, I think this is a good idea and I can do a followup PR that adds this function and switches the existing code to use it, but I also think it's a good idea to keep this PR as simple as possible. thoughts?

Reply:
ok.